pkg/controller/beat/common/pod.go

```diff
@@ -146,11 +204,34 @@ func buildPodTemplate(
 		WithVolumes(volumes...).
 		WithVolumeMounts(volumeMounts...).
 		WithInitContainers(initContainers...).
-		WithInitContainerDefaults()
```

Comment:
IIUC the `-e` flag passed as an argument prevents the main beat process to write a log file on disk (warning in this section https://www.elastic.co/guide/en/beats/filebeat/current/configuration-logging.html#configuration-logging) despite we have configured the logging settings for this.

Reply:
@thbkrkr I didn't update the comments here, but changes were pushed that move logging to a file when stack monitoring is enabled.